### PR TITLE
Adds an on boarding

### PR DESCRIPTION
**Pull request by chacaa**
This feature adds an on boarding for the app.

## Trelo Card:
* [Onboarding logic](https://trello.com/c/DsnNx69F)
* [Setup on boarding images](https://trello.com/c/PoBK5Xte)

## UI:
UI

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -25,6 +25,11 @@
             android:screenOrientation="portrait"
             />
 
+        <meta-data
```
Comment (mirland) [inline app/src/main/AndroidManifest.xml]: remove the key

File: app/src/main/java/com/xmartlabs/doapp/ui/OnBoardingActivity.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+@HensonNavigable
+public class OnBoardingActivity extends AhoyOnboarderActivity {
+  private List<AhoyOnboarderCard> pages = new ArrayList<>();
+  private ViewPager cardViewPager;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    cardViewPager = (ViewPager) findViewById(R.id.vp_pager);
+    createOnboardingCards("", "Create groups for different purpouses", R.drawable.tutorial_1);
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/doapp/ui/OnBoardingActivity.java]: use string resources

File: app/src/main/java/com/xmartlabs/doapp/ui/OnBoardingActivity.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    createOnboardingCards("", "Add & Manage Tasks", R.drawable.tutorial_2);
+    createOnboardingCards("", "View how you are doing!", R.drawable.tutorial_3);
+    createOnboardingCards("", "Go to dashboard", R.drawable.tutorial_4);
+    setImageBackground(R.drawable.bg_walkthrough);
+    showNavigationControls(false);
+    setOnboardPages(pages);
+    setFinishButtonTitle("Let's Do it!");
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/doapp/ui/OnBoardingActivity.java]: too

File: app/src/main/java/com/xmartlabs/doapp/ui/OnBoardingActivity.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    cardViewPager.setPadding(75, 276, 75, 281);
+  }
+
+  @Override
+  public void onFinishButtonPressed() {
+    //TODO go to dashboard
+  }
+
+  private void createOnboardingCards(String title, String description, @DrawableRes int iconId) {
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/doapp/ui/OnBoardingActivity.java]: all titles are empty, so remove it

File: app/src/main/java/com/xmartlabs/doapp/ui/StartActivity.java [hunk 5]
```diff
@@ -35,4 +35,4 @@ protected void onCreate(Bundle savedInstanceState) {
     intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
     getContext().startActivity(intent);
   }
-}
+}
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/doapp/ui/StartActivity.java]: add end line

File: app/src/main/res/values/colors.xml [hunk 6]
```diff
@@ -17,5 +17,5 @@
     <color name="pink">#ff4081</color>
     <color name="white">@android:color/white</color>
     <color name="reddish_pink">#ff3366</color>
-    
+    <color name="dark">#1d1d26</color>
```
Comment (mirland) [inline app/src/main/res/values/colors.xml]: change the color name, `color_dark`

File: app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -48,5 +53,10 @@
             android:screenOrientation="portrait"
             />
 
+        <activity
+            android:name="com.xmartlabs.doapp.ui.OnBoardingActivity"
```
Comment (mirland) [inline app/src/main/AndroidManifest.xml]: shouldn't be single task?

File: app/src/main/java/com/xmartlabs/doapp/ui/OnBoardingActivity.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+import com.xmartlabs.template.R;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by scasas on 3/13/17.
+ */
+@HensonNavigable
+public class OnBoardingActivity extends AhoyOnboarderActivity {
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/doapp/ui/OnBoardingActivity.java:20]: I don't like it, does not it support fragments?

**Review by mirland — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/xmartlabs/doapp/ui/OnBoardingActivity.java (hunk 8 above)
Comment (chacaa) [reply app/src/main/java/com/xmartlabs/doapp/ui/OnBoardingActivity.java:20]: It doesn't :(

File: app/src/main/AndroidManifest.xml [hunk 9]
```diff
@@ -46,16 +41,19 @@
...
         <activity
             android:name="com.xmartlabs.doapp.ui.SingupActivity"
             android:screenOrientation="portrait"
+            android:launchMode="singleTask"
             />
 
         <activity
             android:name="com.xmartlabs.doapp.ui.OnBoardingActivity"
             android:screenOrientation="portrait"
+            android:launchMode="singleTask"
```
Comment (mirland) [inline app/src/main/AndroidManifest.xml]: alpha

File: app/src/main/res/values/colors.xml [hunk 10]
```diff
@@ -17,5 +17,5 @@
     <color name="pink">#ff4081</color>
     <color name="white">@android:color/white</color>
     <color name="reddish_pink">#ff3366</color>
-    <color name="dark">#1d1d26</color>
+    <color name="color_dark">#1d1d26</color>
```
Comment (mirland) [inline app/src/main/res/values/colors.xml]: haha my bad, the color name should be something like `black_dark` or `blue_dark`. I don't know what color is `#1d1d26`

**Review by mirland — CHANGES_REQUESTED** [review]

**Review by mirland — APPROVED** [review]

**Comment by bryant1410** [comment]
Just a comment: it's hard for background pictures to have the right size, as devices have different screen sizes. One possibility is to use "nodpi" with just one image and centered, and you see whatever fits.

**Pull request merged**